src/Ducks/DeleteAccount/Sagas.ts

```diff
@@ -0,0 +1,80 @@
+import { Toast } from 'native-base';
+import { all, takeLatest, call, select } from 'redux-saga/effects';
+
+import api from '../../Services/Api';
+
+import { CentralNavigationService } from '../../Services/Navigation';
+import { AppStackParamList } from '../../types';
+import { root } from '../types';
+
+import { DeleteAccountActionTypes } from './types';
+
+export const getClientId = (state: root): string => state.login.clientId;
+
+export function* validatePassword(action: any) {
+  const clientId: string = yield select(getClientId);
+  const password: string = action.payload;
+
+  try {
+    yield call(api().post, `authentication/${clientId}/validate-password`, password);
+    yield call(deleteAccount, clientId);
+  } catch (error) {
+    Toast.show({
+      text: error?.response?.data?.message ?? 'Erro genérico!'
+    });
+  }
+}
+
+export function* deleteAccount(clientId: string) {
+  try {
+    const accountPost = yield call(api('8082').get, `/clients/${clientId}/accounts`);
+    const accountId: string = accountPost.data[0].accountId;
+    yield call(api('8082').delete, `/clients/${clientId}/accounts/${accountId}`);
```

Comment:
Should Navigate to login after this. 

Maybe we need to create an action to clean Client's Reducer and call it after using this API.